plugins/ecs-bridge/engine/engine.go

```diff
@@ -217,38 +220,59 @@ func (engine *engine) ConfigureContainerVethInterface(netnsName string, result *
 
 // ConfigureBridge configures the IP address of the bridge if needed
 func (engine *engine) ConfigureBridge(result *current.Result, bridge *netlink.Bridge) error {
-	addrs, err := engine.netLink.AddrList(bridge, syscall.AF_INET)
-	if err != nil && err != syscall.ENOENT {
-		return errors.Wrapf(err,
-			"bridge configure: unable to list addresses for bridge %s",
-			bridge.Attrs().Name)
-	}
-
 	resultBridgeNetwork := &net.IPNet{
 		IP:   result.IPs[0].Gateway,
 		Mask: result.IPs[0].Address.Mask,
 	}
 	resultBridgeCIDR := resultBridgeNetwork.String()
-	if len(addrs) > 0 {
-		for _, addr := range addrs {
-			if addr.IPNet.String() == resultBridgeCIDR {
-				return nil
-			}
-		}
-		return errors.New("bridge configure: mismatch in bridge ip address")
+
+	exists, err := engine.bridgeNetworkExists(bridge, resultBridgeCIDR)
+	if err != nil {
+		return err
+	}
+
+	if exists {
+		return nil
 	}
 
 	bridgeAddr := &netlink.Addr{
 		IPNet: resultBridgeNetwork,
 	}
 	err = engine.netLink.AddrAdd(bridge, bridgeAddr)
-	if err != nil {
-		return errors.Wrapf(err,
-			"bridge configure: unable to assign ip address to bridge %s",
+	if err == nil {
+		return nil
+	}
+
+	// if we fail to assign ip address, check bridge network exists or not again, because the error may be caused by
+	// someone else assigned the address right before we assigned it
+	exists, _ = engine.bridgeNetworkExists(bridge, resultBridgeCIDR)
+	if exists {
+		return nil
+	}
+
+	return errors.Wrapf(err,
```

Comment:
This is potentially a bug since you've already dealt with error in line 242. Please move this up to line 242 as a `if err != nil`  block.

Reply:
to fix the race condition, we need to proceed to bridgeNetworkExists even if AddrAdd returns error in line 242, so i don't think we can return error after AddrAdd..

Reply:
Can you use a different variable name (than `err`) to store the return value in line 241? If someone adds any other logic in between lines 242 and 253 in the future and uses `err` to store the return value, this logic will become buggy.

Reply:
updated.